v2/processor.go

```diff
@@ -82,23 +83,55 @@ func NewProcessor(receiver Receiver, handler HandlerFunc, options *ProcessorOpti
 		}
 	}
 	return &Processor{
-		receiver:          receiver,
+		receivers:         []Receiver{receiver},
 		handle:            handler,
 		options:           opts,
 		concurrencyTokens: make(chan struct{}, opts.MaxConcurrency),
 	}
 }
 
-// Start starts the processor and blocks until an error occurs or the context is canceled.
+func (p *Processor) AddReceiver(receivers ...Receiver) {
+	p.receivers = append(p.receivers, receivers...)
+}
+
+// Start starts processing on all the receivers of the processor and blocks until all processors are stopped or the context is canceled.
+// It will retry starting the processor based on the StartMaxAttempt and StartRetryDelayStrategy.
+// Returns a combined list of errors encountered during each processor start.
+func (p *Processor) Start(ctx context.Context) error {
+	wg := sync.WaitGroup{}
+	errChan := make(chan error, len(p.receivers))
+	for i := range p.receivers {
+		wg.Add(1)
+		go func(index int) {
+			defer wg.Done()
+			err := p.StartAtIndex(ctx, index)
+			if err != nil {
+				errChan <- err
+			}
+		}(i)
+	}
+	wg.Wait()
+	close(errChan)
+	var allErrs []error
+	for err := range errChan {
+		allErrs = append(allErrs, err)
+	}
+	return errors.Join(allErrs...)
+}
+
+// StartAtIndex starts the processor with receiver index and blocks until an error occurs or the context is canceled.
 // It will retry starting the processor based on the StartMaxAttempt and StartRetryDelayStrategy.
 // Returns a combined list of errors during the start attempts or ctx.Err() if the context
 // is cancelled during the retries.
-func (p *Processor) Start(ctx context.Context) error {
+func (p *Processor) StartAtIndex(ctx context.Context, index int) error {
```

Comment:
why do we need this api? not fond of exposing the index.
how does the user know which receiver is which index? especially with the AddReceiver api?